### PR TITLE
Feature/1.6/discovery/state

### PR DESCRIPTION
**Pull request by lazan**
Probably best to review commit by commit.

### CODE REVIEW

File: middleware/domain/include/domain/discovery/state.h [hunk 1]
```diff
@@ -23,17 +23,114 @@ namespace casual
...
+
+                  template< count::Tag tag, typename C, typename... Messages>
+                  constexpr auto counts( C creator, Messages... messages) noexcept
+                  {
+                     using value_type = decltype( creator( common::message::Type{}, platform::size::type{}));
+
+                     std::vector< value_type> result;
+                     result.reserve( sizeof...( Messages));
+
+                     ( ... , result.push_back( detail::create< tag>( creator, messages)) );
```
Comment (AlbsAgren) [inline middleware/domain/include/domain/discovery/state.h:78]: How does this work? I assume it unpacks into a push_back for each Messages parameter but the syntax is completely new to me.

Comment (lazan) [reply middleware/domain/include/domain/discovery/state.h:78]: A feature in c++17, fold-expressions https://en.cppreference.com/w/cpp/language/fold. Not often that you need it, but when you do, it's awesome!

File: middleware/domain/source/discovery/admin/cli.cpp [hunk 2]
```diff
@@ -44,6 +70,243 @@ namespace casual
...
+               namespace providers
+               {
+                  auto option()
+                  {
+                     static constexpr auto create_formatter = []()
+                     {
+                        auto format_pid = []( auto& provider) { return provider.process.pid;};
+                        auto format_abilities = []( auto& provider) { return provider.abilities;};
+                        
+                        //auto format_alias = []( auto& i) { return i.server->alias;};
```
Comment (AlbsAgren) [inline middleware/domain/source/discovery/admin/cli.cpp:85]: Leftover here

**Review by AlbsAgren — APPROVED** [review]

**Comment by lazan** [comment]
Fixed the pr-comment locally and pushed.

**Pull request closed without merging**